### DISCUSSION TITLE
Is it possible to use django-components with Jinja backend?

### DISCUSSION THREAD

**Discussion by elied** [django-components/django-components#301] [Ideas]
I came across django-components and am very interested to give it a try, but my team uses Django's Jinja2 backend and I'm having trouble making things work.

I'm far from an expert with Django's templating settings, but there seem to be a few issues. Firstly, I don't know how to use the django_components.template_loader.Loader, since the Jinja backend seems to only support a single 'loader' option ([docs](https://docs.djangoproject.com/en/4.2/topics/templates/#django.template.backends.jinja2.Jinja2)).

Despite that, I naively tried to define the 'component' templatetag as an environment filter, but that unsurprisingly led to other errors.

Is there a way to get this working? Maybe the [django-jinja](https://niwi.nz/django-jinja/latest/) would make this easier?

**Comment by EmilStenstrom** [comment]
@elied Hi! I’m afraid I haven’t tried it with Jinja, and I don’t think I’ve seen anyone else asking about it here either. I’m open to PRs adding support for Jinja given it’s not too much of a maintenance burden. Maybe you could give it a try and report back?

**Reply by elied** [reply]
I didn't get this working, though I also didn't try too much harder. It's very possible that other Jinja-users could shim something together that just calls into some of your components code pretty easily, but I don't anticipate it'd be something you'd want to officially add to the repo. So while I really love the concept and the work you put into it, I think Jinja-users are going to want to look elsewhere :)

For future readers, I just came across JinjaX (https://jinjax.scaletti.dev/guide/) which is overall similar to django-components but is obviously designed for Jinja.

**Reply by EmilStenstrom** [reply]
Happy you found another solution!

**Comment by rtr1** [comment]
@EmilStenstrom there is some discussion about making Jinja the default templating language by the Django team.

From the Django Newsletter:

> Move to Jinja2 as the default template language

> This is a big one, but I believe it's doable given a few improvements to template backend selection. The main barrier will be all the admin templates, as you cannot (currently?) extend or include templates across engines.

I assume DTL would remain, but something to keep an eye on.

https://docs.google.com/document/d/11Z0g7YXo1XveHW4XQnFqVo0o-4rkv-FxrSgxdMWGfXk/mobilebasic

**Reply by EmilStenstrom** [reply]
I'm open to discuss what's needed to add optional support for Jinja. I worry that it's a lot of work, since the core of this library is a couple of template tags, likely specific to the Django Template Language.

**Comment by JuroOravec** [comment]
Yeah, Jinja2 will definitely be a lot of work. At first I was playing with the idea, but currently we're tightly coupled to DTL.

@elied, @rtr1, or others, if you have some public projects that use Django with Jinja2, could you link them here? The major block for me is that I never used Django + Jinja2 setup, so I don't know how it should work and what would people expect.
